packages/rocketchat-lib/server/lib/sendEmailOnMessage.js

```diff
@@ -4,7 +4,7 @@ import s from 'underscore.string';
 function getEmailContent({ messageContent, message, user, room }) {
 	const lng = user && user.language || RocketChat.settings.get('language') || 'en';
 
-	const roomName = `#${ RocketChat.settings.get('UI_Allow_room_names_with_special_chars') ? room.fname || room.name : room.name }`;
+	const roomName = `#${ (RocketChat.settings.get('UI_Allow_room_names_with_special_chars') && room.fname) ? room.fname : RocketChat.roomTypes.getRoomName(room.t, room) }`;
```

Comment:
it is not needed anymore to test for  `UI_Allow_room_names_with_special_chars` setting here. this test is done on implementations of `getRoomName` (like [this](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-lib/lib/roomTypes/public.js#L37))..